crates/pop-cli/src/commands/clean.rs

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::{cache, style::Theme};
+use anyhow::Result;
+use clap::{Args, Subcommand};
+use cliclack::{clear_screen, confirm, intro, log, multiselect, outro, outro_cancel, set_theme};
+use console::style;
+use std::{
+	fs::{read_dir, remove_file},
+	path::PathBuf,
+};
+
+#[derive(Args)]
+#[command(args_conflicts_with_subcommands = true)]
+pub(crate) struct CleanArgs {
+	#[command(subcommand)]
+	pub(crate) command: Command,
+}
+
+/// Remove generated/cached artifacts.
+#[derive(Subcommand)]
+pub(crate) enum Command {
+	#[cfg(feature = "parachain")]
+	/// Remove cached artifacts.
+	#[clap(alias = "c")]
+	Cache(CleanCacheCommand),
+}
+
+#[derive(Args)]
+pub(crate) struct CleanCacheCommand;
+
+impl CleanCacheCommand {
+	/// Executes the command.
+	pub(crate) fn execute(self) -> Result<()> {
+		clear_screen()?;
+		set_theme(Theme);
+		intro(format!("{}: Remove cached artifacts", style(" Pop CLI ").black().on_magenta()))?;
+
+		// Get the cache contents
+		let cache = cache()?;
+		if !cache.exists() {
+			outro_cancel("🚫 The cache does not exist.")?;
+			return Ok(());
+		};
+		let contents = contents(&cache)?;
+		if contents.is_empty() {
+			outro(format!(
+				"ℹ️ The cache at {} is empty.",
+				cache.to_str().expect("expected local cache is invalid")
+			))?;
+			return Ok(());
+		}
+		log::info(format!(
+			"ℹ️ The cache is located at {}",
+			cache.to_str().expect("expected local cache is invalid")
+		))?;
+
+		// Prompt for selection of artifacts to be removed
+		let mut prompt = multiselect("Select the artifacts you wish to remove:").required(false);
+		for (name, path, size) in &contents {
+			prompt = prompt.item(path, name, format!("{}MiB", size / 1_048_576))
+		}
+		let selected = prompt.interact()?;
+		if selected.is_empty() {
+			outro("ℹ️ No artifacts removed")?;
+			return Ok(());
+		}
+
+		// Confirm removal
+		let prompt = match selected.len() {
+			1 => "Are you sure you want to remove the selected artifact?".into(),
+			_ => format!(
+				"Are you sure you want to remove the {} selected artifacts?",
+				selected.len()
+			),
+		};
+		if !confirm(prompt).interact()? {
+			outro("ℹ️ No artifacts removed")?;
+			return Ok(());
+		}
+
+		// Finally remove selected artifacts
+		for file in &selected {
+			remove_file(file)?
```

Comment:
If it is a folder, it does not remove it.

Reply:
Do we need to remove folders? I dont expect there to be any at this moment in time, so perhaps the listing should only show files and exclude folders?

Reply:
When extracting the contents for the `substrate-contracts-node`, it includes the binary inside a folder named "artifacts": https://github.com/r0gue-io/pop-cli/blob/main/crates/pop-contracts/src/utils/contracts_node.rs#L59

Reply:
Not saying this is the solution, but would it not be better to move it out of the archive folder on download? https://github.com/r0gue-io/pop-cli/blob/03aecfe8b684b8a0004158eb1d14d081e7253ee2/crates/pop-parachains/src/up/sourcing.rs#L230

Otherwise the implementation here is going to get more complicated. `artifacts` has no real meaning, requiring nested traversal to collect all files in all subdirectories, so we would need to prefix the name with the relative path.

Reply:
Also, not sure if the contracts node checks for newer versions and prompts for updates but if not, might be worthwhile moving the sourcing implementations to a common crate to maximise reuse and consistency.

Reply:
It makes sense to have the same behaviour for contracts-node. At least the first point of removing the folder when downloading. Opened and issue: https://github.com/r0gue-io/pop-cli/issues/224